contrib/epee/src/hex.cpp

```diff
@@ -84,7 +86,42 @@ namespace epee
     return write_hex(out, src);
   }
 
-  std::vector<uint8_t> from_hex::vector(boost::string_ref src)
+
+  bool from_hex::to_string(std::string& out, const boost::string_ref src)
+  {
+    out.resize(src.size() / 2);
+    return to_buffer_unchecked(reinterpret_cast<std::uint8_t*>(&out[0]), src);
+  }
+
+  bool from_hex::to_buffer(span<std::uint8_t> out, const boost::string_ref src) noexcept
+  {
+    if (src.size() / 2 != out.size())
```

Comment:
Wouldn't it be nicer to multiple out.size() by 2 to compare instead of doing integer division? If `src.size()` is odd, this would continue to the `to_buffer_unchecked` before it is rejected.

Reply:
Although unlikely, multiplication by 2 can overflow.

Reply:
That's true, but since they're both signed it's perfectly defined behaviour. And since both operands are of the same type (`std::size_t`) if the multiplication overflows the division wouldn't compare the same either.

I do get your point, though. Also, the current code does of course work correctly.

Reply:
> but since they're both signed

I think you meant `unsigned`? I don't think multiplication works for all cases. One (again, extreme example), would be if `out.size() == (size_t(1) << 63) | size_t(1)` and if `src.size() == 2`. Then `out.size() * 2 == src.size()`, and `src` is still divisible by `2` so the check in `to_buffer_unchecked` succeeds. This still has defined behavior since `out` is larger than `src`, but its no longer rejecting some inexact `out.size()`s.

This is admittedly a contrived example, but the division by 2 is going to be a right shift by 1 in ASM. So the only penalty is sometimes an additional bitmask and comparison is performed if the division check passes.